cmd/cloud-run/rotate-signify-certificate/rotate_signify_certificate.py

```diff
@@ -0,0 +1,279 @@
+    )
+
+    return {
+        "certData": base64.b64encode(certs_string.encode()).decode(),
+        "privateKeyData": base64.b64encode(private_key_bytes).decode(),
+        "createdAt": created_at,
+        "certServiceURL": secret_data["certServiceURL"],
+        "tokenURL": secret_data["tokenURL"],
+        "clientID": secret_data["clientID"],
+        "password": "",  # keep it empty to maintain structure, but indicate it's not needed
+    }
+
+
+def extract_message_data(pubsub_message: Any) -> Any:
+    """Extract secret rotate emssage from pubsub message"""
+    if pubsub_message["attributes"]["eventType"] != "SECRET_ROTATE":
+        # pylint: disable=broad-exception-raised
+        raise Exception("Unsupported event type")
+
+    data = base64.b64decode(pubsub_message["data"])
+    return json.loads(data)
+
+
+def decrypt_private_key(private_key_data: bytes, password: bytes) -> bytes:
+    """Decrypr encrypted private key"""
+    # pylint: disable=line-too-long
+    private_key = serialization.load_pem_private_key(private_key_data, password)
+
+    # pylint: disable=line-too-long
+    return private_key.private_bytes(
+        Encoding.PEM, PrivateFormat.PKCS8, serialization.NoEncryption()
+    )
+
+
+def fetch_access_token(
+    certificate: bytes, private_key: bytes, token_url: str, client_id: str
+) -> str:
+    """fetches access token from given token_url using certificate and private key"""
+    # Use temporary file for old cert and key because requests library needs file paths,
+    # it's not a security concern because the code is running in known environment controlled by us
+    # pylint: disable=line-too-long
+    with tempfile.NamedTemporaryFile() as old_cert_file, tempfile.NamedTemporaryFile() as old_key_file:
+
+        old_cert_file.write(certificate)
+        old_cert_file.flush()
+
+        old_key_file.write(private_key)
+        old_key_file.flush()
+
+        # pylint: disable=line-too-long
+        access_token_response = requests.post(
+            token_url,
+            cert=(old_cert_file.name, old_key_file.name),
+            data={
+                "grant_type": "client_credentials",
+                "client_id": client_id,
+            },
+            timeout=30,
+        ).json()
+
+        return access_token_response["access_token"]
+
+
+# TODO(kacpermalachowski): Move it to common package
+def prepare_log_fields() -> Dict[str, Any]:
+    """prepare_log_fields prapares basic log fields"""
+    log_fields: Dict[str, Any] = {}
+    request_is_defined = "request" in globals() or "request" in locals()
+    if request_is_defined and request:
+        trace_header = request.headers.get("X-Cloud-Trace-Context")
+        if trace_header and project_id:
+            trace = trace_header.split("/")
+            log_fields["logging.googleapis.com/trace"] = (
+                f"projects/{project_id}/traces/{trace[0]}"
+            )
+    log_fields["Component"] = "signify-certificate-rotator"
```

Comment:
```suggestion
    log_fields["component"] = "signify-certificate-rotator"
```
can it be lower case as other entries?

Reply:
I created the issue to share that code, I prefer to keep it similar to slack-message-sender one: https://github.com/orgs/kyma-project/projects/22/views/51?pane=issue&itemId=79537174